src/callbacks.cpp

```diff
@@ -0,0 +1,231 @@
+#include "main.hpp"
+
+/**
+ * @brief Callback function for the control message.
+ *
+ * This function is called when a control message is received. It processes the
+ * message, activates the node based on the message, parses the payload, and
+ * saves the data in a Command struct.
+ *
+ * @param msg The control message received.
+ */
+void WaypointNode::callback_control(const interfaces::msg::Control &msg) {
+    // Ignore messages that don't target this node
+    if (msg.target_id != this->get_name()) {
+        RCLCPP_DEBUG(this->get_logger(),
+                     "WaypointNode::%s: Ignored control message, "
+                     "as it was not targeted for this node",
+                     __func__);
+        return;
+    }
+
+    // Init node state
+    reset_node();
+
+    // Activate or deactivate node based on the message
+    if (msg.active != this->get_active()) {
+        if (msg.active) {
+            this->activate();
+
+            RCLCPP_INFO(this->get_logger(),
+                        "WaypointNode::%s: --- NODE ACTIVATED ---", __func__);
+        } else {
+            this->deactivate();
+
+            RCLCPP_INFO(this->get_logger(),
+                        "WaypointNode::%s: --- NODE DEACTIVATED ---", __func__);
+        }
+    }
+
+    // If the node should be deactivated, don't parse the payload as it is only
+    // relevant in active mode
+    if (!this->get_active()) {
+        return;
+    }
+
+    // Parse payload
+    nlohmann::json cmd_json;
+    try {
+        cmd_json = common_lib::CommandDefinitions::parse_check_json_str(
+            msg.payload,
+            common_lib::CommandDefinitions::get_waypoint_command_definition());
+    } catch (const std::runtime_error &e) {
+        RCLCPP_FATAL(this->get_logger(),
+                     "WaypointNode::%s: Received invalid json as "
+                     "payload: %s",
+                     __func__, e.what());
+        this->job_finished(
+            (std::string) "WaypointNode::" + (std::string) __func__ +
+            ": Received "
+            "invalid json as payload: " +
+            e.what());
+        reset_node();
+        return;
+    }
+
+    // Set flag to true to indicate that the object has values in it
+    cmd.values_set = true;
+
+    // Required parameters
+    cmd.target_coordinate_lat = cmd_json.at("target_coordinate_lat");
```

Comment:
wouldn´t it be better to store these important keys inside common lib or the CommandDefiniton.
Now the check could pass, but if you made a mistake in the string there would be a failure at this point.

Reply:
I get your point and these keys are indeed already stored in the common lib.
But accessing them would require me to know the string anyway, so that wouldn't solve anything.
All in all, I don't see that as a problem as the contents are well defined, documented, and most importantly stable.